### ISSUE TITLE
I get Internal Error when I check the URL it should generates

### ISSUE THREAD

**Issue by toiluj23** [juice49/next-og-image#3]
Hi, I followed your steps, but perhaps im doing something wrong and it's not an issue. You let me know.

This is the URL I want to check:
http://localhost:3000/products

This should be the URL I need to build:
http://localhost:3000/api/og-image/products/og-image.png

**I've generated the following structure:** 
api / og-image / [...path].ts

I've generated the following var in my .env:
OG_IMAGE_BASE_URL=http://localhost:3000

This is what I get:
![image](https://user-images.githubusercontent.com/1655880/118676510-d97fa080-b7d1-11eb-9920-b606098e89e1.png)

PS: thanks for adding a library to do this, i believe it's super valuable.
PS2: I case I make it works, is it possible to modify the design of the OG Image?

**Comment by juice49** [comment]
Hi @toiluj23,

I think the internal error is because you are running the project locally. We use the [chrome-aws-lambda](https://github.com/alixaxel/chrome-aws-lambda/wiki/HOWTO:-Local-Development) package, and it currently only works when deployed to AWS Lambda. I have opened issue #4 for this, because we should at least mention it in the readme.

A good way to test your OG image would be to deploy your project to [Vercel](https://vercel.com) (because Vercel serverless functions are deployed to AWS Lambda). Everything should run as expected on Vercel. You will need to remove or comment out the `OG_IMAGE_BASE_URL` environment variable before deploying, because Next OG Image will automatically use the URL of your Vercel deployment.

> PS2: I case I make it works, is it possible to modify the design of the OG Image?

Yes, absolutely 🙂. Next OG Image works by creating a snapshot of a webpage, so you can create a page to template your OG image and make it look however you want. For your project, you would need to create two Next pages:

1. `pages/products/index.tsx`: the products page on your website.
2. `pages/products/og-image.tsx`: a dedicated page to act as a template for your products OG image.

When you navigate to `api/og-image/products/og-image.png`, Next OG Image will create and serve a snapshot of the page `pages/products/og-image.tsx` page.

I've created [a demo project](https://github.com/juice49/next-og-image-example) to help. The demo contains two pages:

1. [`pages/about/index.tsx`](https://github.com/juice49/next-og-image-example/blob/main/pages/about/index.tsx): an About page.
2. [`pages/about/og-image.tsx`](https://github.com/juice49/next-og-image-example/blob/main/pages/about/og-image.tsx): a dedicated page to act as a template for the about page OG image. You can put any HTML and CSS you want here to template the OG image.

You can see this working on Vercel:

1. [The About page](https://next-og-example-ash1.vercel.app/about).
2. [The About page OG image template](https://next-og-example-ash1.vercel.app/about/og-image).
3. [Rendered About page OG image](https://next-og-example-ash1.vercel.app/api/about/og-image.png).

My personal site might be a better example of the kind of templating you can do:

1. [A track page](https://ash.gd/this-is-my-jam/b1683db8-5d7b-494b-be3f-61a4e6c5f8d0).
2. [Track page OG image template](https://ash.gd/this-is-my-jam/b1683db8-5d7b-494b-be3f-61a4e6c5f8d0/og-image).
3. [Rendered track page OG image](https://og-image.ash.gd/api/this-is-my-jam/b1683db8-5d7b-494b-be3f-61a4e6c5f8d0/og-image.png).

I hope this helps. Sorry the documentation isn't great currently. Let me know if you have any other questions 🙂.

N.B. I am also working on issue #5 to help make templating more more flexible. This would allow you to pass props to the template via the URL, for example: `?title=Blog%20Post%20Title&author=Ash`.

**Comment by toiluj23** [comment]
Thanks, I will check this. And I have it on Vercel, as soon as I can deploy it, i will show you the results. :)

**Comment by toiluj23** [comment]
Dear @juice49 I've did all the steps and wanted to upload to vercel, but this is the LOG, looks like a different issue but i wanted to share it here.

Take a look.
![image](https://user-images.githubusercontent.com/1655880/118892926-061de000-b8d8-11eb-8b73-1c2b513f6e0b.png)

**Comment by juice49** [comment]
@toiluj23 I encountered the same issue. I think it's related to this behaviour of Next.js:

> NOTE: Next.js bundles all API Routes and server-rendered pages into shared Serverless Functions , separating API routes and server-rendered pages when deployed on Vercel. Further Serverless Functions are created only if they do not fit within the 50MB limit.

Source: [Vercel limits](https://vercel.com/docs/platform/limits#general-limits).

It seems like something strange is going on. Disabling Webpack 5 fixed the problem for me.

For my project, I wanted to use Webpack 5, so instead I created a [separate Vercel project](https://github.com/juice49/personal-site-og-image) solely for the purpose of hosting Next OG Image. Because there are no other API functions in the dedicated project, it can fit into the 50mb limit.

Not an ideal fix, but it worked for now.
